Bottom: a7f8cbe4a0c625153e7d2ce125d67e7e96527528
Top:    e8d755bc3c5aba482211eec2d2229d8ad1d722bd
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2008-10-02 19:16:41 +0100

Workaround crash where contour is passed as NULL to poly_PreContour


---

diff --git a/src/polygon.c b/src/polygon.c
index c33c735..54f8f9f 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -202,6 +202,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
